src/main/java/reposense/model/RepoConfiguration.java

```diff
@@ -28,11 +40,25 @@
     private transient Map<Author, String> authorDisplayNameMap = new HashMap<>();
     private transient boolean annotationOverwrite = true;
 
-    public RepoConfiguration(String organization, String repoName, String branch) {
-        this.organization = organization;
-        this.repoName = repoName;
+
+    /**
+     * @throws ParseException if {@code location} cannot be represented by a {@code URL} or {@code Path}.
+     */
+    public RepoConfiguration(String location, String branch) throws ParseException {
+        this.location = location;
         this.branch = branch;
-        this.displayName = organization + "_" + repoName + "_" + branch;
+
+        verifyLocation(location);
+        Matcher matcher = GIT_REPOSITORY_LOCATION_PATTERN.matcher(location);
+
+        if (matcher.matches()) {
+            String organization = matcher.group("org");
+            repoName = matcher.group("repoName");
+            displayName = organization + "_" + repoName + "_" + branch;
+        } else {
+            repoName = Paths.get(location).getFileName().toString();
+            displayName = repoName + "_" + branch;
```

Comment:
Hmm, this may give rise to the same problem mentioned in #206, whereby we give 2 different locations but the folder name is same, and use the same branch, end up overriding each other's report.
Perhaps we can temporarily set the repoName to be the absolute path, just replace the file separators with `_`?

Reply:
Hmm, is `getFileName()` correct in the first place? Aren't we making user entering `.git` at the end by default?

Reply:
Will open a issue for this.